[BUG]: fetchcontent started failing in combination with NVHCP

### Is this a duplicate?

- [x] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cccl/issues) for this bug and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Type of Bug

Compile-time Error

### Component

Thrust

### Describe the bug

When pulling CCCL through Cmake's Fetchcontent, I started getting this message:

```
-- Git command found: /usr/bin/git
-- Finding CCCL components: Thrust;CUB;libcudacxx
-- Found CUB: /builds/correaa/boost-multi/build/_deps/cccl-cmake-src/lib/cmake/cub/cub-config.cmake (found suitable version "3.3.0.0", minimum required is "3.3.0.0") 
CMake Error at build/_deps/cccl-cmake-src/lib/cmake/libcudacxx/libcudacxx-config.cmake:158 (define_property):
  define_property not given a BRIEF_DOCS <brief-doc> argument.
Call Stack (most recent call first):
  build/_deps/cccl-cmake-src/lib/cmake/libcudacxx/libcudacxx-config.cmake:245 (libcudacxx_update_language_compat_flags)
  build/_deps/cccl-cmake-src/lib/cmake/thrust/thrust-config.cmake:943 (find_package)
  build/_deps/cccl-cmake-src/lib/cmake/cccl/cccl-config.cmake:77 (find_package)
  build/_deps/cccl-cmake-src/cmake/CCCLAddSubdirHelper.cmake:64 (include)
  build/_deps/cccl-cmake-src/cmake/CCCLAddSubdir.cmake:1 (cccl_add_subdir_helper)
  build/_deps/cccl-cmake-src/CMakeLists.txt:24 (include)
```

All the details are here:
https://gitlab.com/correaa/boost-multi/-/jobs/12307373426#L222

This is the code in CmakeLists.txt:

```
            include(FetchContent)
            FetchContent_Declare(
                cccl-cmake
                GIT_REPOSITORY https://github.com/NVIDIA/cccl.git
                GIT_TAG main
                GIT_SHALLOW ON
            )
            FetchContent_MakeAvailable(cccl-cmake)
```

### How to Reproduce

In a nvhpc system (e.g. 23.11)

```
$ git clone https://gitlab.com/correaa/boost-multi.git
$ cd boost-multi
$ mkdir build; cd build
$ CXX=/opt/nvidia/hpc_sdk/Linux_x86_64/2023/compilers/bin/nvc++ cmake ..
```

### Expected behavior

cmake should pull and configure CCCL

### Reproduction link

_No response_

### Operating System

_No response_

### nvidia-smi output

_No response_

### NVCC version

_No response_

--- a/lib/cmake/libcudacxx/libcudacxx-config.cmake
+++ b/lib/cmake/libcudacxx/libcudacxx-config.cmake
@@ -112,6 +112,16 @@ target_compile_definitions(
   INTERFACE $<$<CONFIG:Debug>:CCCL_ENABLE_ASSERTIONS>
 )
 
+function(_libcudacxx_define_internal_global_property prop_name)
+  # Need to define docs options for CMake < 3.23 (optional in later versions)
+  define_property(
+    GLOBAL
+    PROPERTY ${prop_name}
+    BRIEF_DOCS "Internal libcudacxx property: ${prop_name}."
+    FULL_DOCS "Internal libcudacxx property: ${prop_name}."
+  )
+endfunction()
+
 # We cannot test for MSVC version if the CXX or CUDA languages aren't enabled, because
 # the CMAKE_[CXX|CUDA_HOST]_COMPILER_[ID|VERSION] variables won't exist.
 # Just call find_package(libcudacxx) again after enabling languages to rediscover.
@@ -146,7 +156,7 @@ function(libcudacxx_update_language_compat_flags)
     message(VERBOSE "libcudacxx:   /Zc:__cplusplus and /Zc:preprocessor flags may not be automatically added to CXX targets.")
     message(VERBOSE "libcudacxx:   Call find_package(CCCL) again after enabling CXX to enable compatibility flags.")
     # gersemi: on
-    define_property(GLOBAL PROPERTY _libcudacxx_cxx_warned)
+    _libcudacxx_define_internal_global_property(_libcudacxx_cxx_warned)
   endif()
 
   if (NOT cuda_warned AND NOT CUDA IN_LIST langs)
@@ -155,7 +165,7 @@ function(libcudacxx_update_language_compat_flags)
     message(VERBOSE "libcudacxx:   /Zc:__cplusplus and /Zc:preprocessor flags may not be automatically added to CUDA targets.")
     message(VERBOSE "libcudacxx:   Call find_package(CCCL) again after enabling CUDA to enable compatibility flags.")
     # gersemi: on
-    define_property(GLOBAL PROPERTY _libcudacxx_cuda_warned)
+    _libcudacxx_define_internal_global_property(_libcudacxx_cuda_warned)
   endif()
 
   if (CXX IN_LIST langs)
@@ -190,7 +200,7 @@ function(libcudacxx_update_language_compat_flags)
       # Usually a safe assumption but provide an escape hatch for edge cases.
       message(STATUS "libcudacxx: - Assuming CUDA host compiler is the same as CXX compiler.")
       message(STATUS "libcudacxx:   Set libcudacxx_MISMATCHED_HOST_COMPILER=TRUE to disable this.")
-      define_property(GLOBAL PROPERTY _libcudacxx_mismatch_warned)
+      _libcudacxx_define_internal_global_property(_libcudacxx_mismatch_warned)
       set(msvc_cuda_host_id ${CMAKE_CXX_COMPILER_ID})
       set(msvc_cuda_host_version ${CMAKE_CXX_COMPILER_VERSION})
     endif()
@@ -221,7 +231,7 @@ function(libcudacxx_update_language_compat_flags)
         )
       endforeach()
     endif()
-    define_property(GLOBAL PROPERTY _libcudacxx_cxx_checked)
+    _libcudacxx_define_internal_global_property(_libcudacxx_cxx_checked)
   endif()
 
   if (NOT cuda_checked AND DEFINED msvc_cuda_host_id)
@@ -238,7 +248,7 @@ function(libcudacxx_update_language_compat_flags)
         )
       endforeach()
     endif()
-    define_property(GLOBAL PROPERTY _libcudacxx_cuda_checked)
+    _libcudacxx_define_internal_global_property(_libcudacxx_cuda_checked)
   endif()
 endfunction()
 

